framework/src/actions/CreateRunnerAction.C

```diff
@@ -0,0 +1,78 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CreateRunnerAction.h"
+#include "Factory.h"
+#include "PetscSupport.h"
+#include "MooseApp.h"
+#include "Runner.h"
+#include "Eigenvalue.h"
+#include "FEProblem.h"
+#include "EigenProblem.h"
+
+registerMooseAction("MooseApp", CreateRunnerAction, "setup_runner");
+
+defineLegacyParams(CreateRunnerAction);
+
+InputParameters
+CreateRunnerAction::validParams()
+{
+  InputParameters params = MooseObjectAction::validParams();
+  params.addClassDescription("Add an Runner object to the simulation.");
+  params.addParam<bool>(
+      "auto_preconditioning",
+      true,
+      "When true and a [Preconditioning] block does not exist, the application will attempt to use "
+      "the correct preconditioning given the Runner settings.");
+  return params;
+}
+
+CreateRunnerAction::CreateRunnerAction(InputParameters params)
+  : MooseObjectAction(params), _auto_preconditioning(getParam<bool>("auto_preconditioning"))
+{
+}
+
+void
+CreateRunnerAction::act()
+{
+  std::shared_ptr<Runner> runner = _factory.create<Runner>(_type, _name, _moose_object_pars);
+
+  // If enabled, automatically create a Preconditioner if the [Preconditioning] block is not found
+  if (_auto_preconditioning && !_awh.hasActions("add_preconditioning") &&
+      _moose_object_pars.isParamValid("solve_type"))
+    setupAutoPreconditioning();
```

Comment:
I thought a runner could be a "flexible" executor that might not have `Preconditioning`

`Preconditioning` might be handled by "NonlinearSystem" ?

Reply:
This is because currently, the runner class is just inheriting from the executioner class - and so copies all the same setup.  This will change eventually.  It's just a matter of determining how and when that transformation occurs.